File: go/http.go
package main

import (
	"fmt"
	"os"
	"time"

	"github.com/go-resty/resty/v2"
)

func Post(method string) {
	PostMethod(Url(method + ".json"))
}

func PostMethod(json map[string]string) {
	url := json["url"]
	body := json["body"]

	client := resty.New()

	client.
		SetTimeout(5 * time.Second)

	res, err := client.R().
		SetHeader("Content-Type", "application/json").
		SetBody(body).
		Post(url)

	Check(url, body, res)
	
	result(res, err)
}

func result(res *resty.Response, err error) {
	if err != nil {
		fmt.Println(err)
		os.Exit(1)
	} else {
		fmt.Print("Status code : ")
		fmt.Println(res.StatusCode())
		fmt.Println("Response : ")
		fmt.Println(JsonPrettyPrint(res.String()))
		if(res.StatusCode() == 500) {
			os.Exit(1)
		}
	}

	// Wait()
}

File: javascript/http.js
const request = require('request');
const log = require('tracer').colorConsole({
  format: '{{message}}',
});

const host = 'http://127.0.0.1:7545/';

exports.post = (method, params) => {
  const options = {
    uri: host + method,
    method: 'POST',
    body: {
      jsonrpc: '2.0',
      method: method,
      params: params,
      id: 1,
    },
    json: true,
  };

  request.post(options, (error, response, body) => {
    if (error != null) {
      log.error('Request: ' + host + method);
      log.error('Params: ' + params);
      log.error(error);
      log.error('\n');
    } else {
      log.info('Request: ' + host + method);
      log.info('Params: ' + params);
      log.info(body);
      log.info('\n');
    }
  });
};
